tests/imageuploadcommand.js

```diff
@@ -78,6 +79,21 @@ describe( 'ImageUploadCommand', () => {
 			expect( getModelData( document ) ).to.equal( `<image src="image.png"></image>[<image uploadId="${ id }"></image>]` );
 		} );
 
+		it( 'should allow inserting image at some custom position', () => {
+			const file = createNativeFileMock();
+			setModelData( document, '[<paragraph>foo</paragraph>]<paragraph>bar</paragraph>' );
+
+			const selectedElement = document.selection.getSelectedElement();
+			const customPosition = ModelPosition.createBefore( selectedElement );
```

Comment:
For the test to be safe (i.e. that we know that it really checks something) the selection position and the custom position should be far away from each other plus the custom position should not be at the any end of the content (so we know that by coincidence the image wasn't inserted at some random start/end position).